yahoo_panoptes/plugins/polling/generic/snmp/plugin_polling_generic_snmp.py

```diff
@@ -0,0 +1,581 @@
+
+        return targets_map, metrics_type_map, top_level_targets_map
+
+    def _add_dimension_to_metrics_group_if_not_empty(self, dimension, value, metrics_group, dimension_was_empty):
+        """
+        Given a dimension and value for a metrics group, add to metrics group if value is not empty. Return the
+        metrics_group and updated dimension_was_empty boolean.
+        """
+        if value != "":
+            metrics_group.add_dimension(metrics.PanoptesMetricDimension(dimension, value))
+        else:
+            dimension_was_empty = True
+
+        return metrics_group, dimension_was_empty
+
+    def _add_top_level_metrics_and_dimensions_to_metrics_group(self, top_level_metrics_map, top_level_dimensions_map,
+                                                               metrics_type_map, metrics_group, dimension_was_empty):
+        """
+        Add all top-level metrics and dimensions in the provided maps to the provided metrics_group.
+        Return updated metrics_group and dimension_was_empty boolean.
+        """
+        for metric in top_level_metrics_map:
+            metrics_group.add_metric(
+                metrics.PanoptesMetric(metric, top_level_metrics_map[metric], metrics_type_map[metric]))
+        # "top_level" metrics don't have indices, so only non-indexed dimensions can be added
+        for dimension, value in top_level_dimensions_map.items():
+            metrics_group, dimension_was_empty = self._add_dimension_to_metrics_group_if_not_empty(dimension,
+                                                                                                   value,
+                                                                                                   metrics_group,
+                                                                                                   dimension_was_empty)
+
+        return metrics_group, dimension_was_empty
+
+    def _add_metrics_group_if_allowed(self, metrics_group, metrics_group_map, metrics_group_name, dimension_was_empty):
+        """
+        Add populated, provided metrics_group to the plugin's PanoptesMetricsGroupSet if no dimension was empty or if
+        'ignore_empty_dimensions flag is True. Otherwise, handle a PanoptesMetricDimensionNullException for the
+        metrics group in the polling status.
+        """
+        if metrics_group_map['ignore_empty_dimensions']:
+            self._metrics.add(metrics_group)
+        elif not dimension_was_empty:
+            self._metrics.add(metrics_group)
+        else:
+            self._polling_status.handle_exception(metrics_group_name, PanoptesMetricDimensionNullException())
+
+    def _process_metrics(self):
+        """
+        If values for the provided oids in the enrichment configuration have been found, then form and populate metrics
+        groups and add to the plugin's PanoptesMetricsGroupSet.
+        """
+        if self._oid_maps or self._snmpget_oid_map:
```

Comment:
If we invert the test (if not self._oid_maps...), then we return statement up top, removing one level of indentation